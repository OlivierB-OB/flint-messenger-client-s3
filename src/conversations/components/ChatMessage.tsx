import React, { Fragment } from 'react';
import { IConversationMessage } from '../types';
import { ListItemText, ListItemAvatar } from '@material-ui/core';
import { ContactName } from '../../users/components/ContactName';
import { ContactAvatar } from '../../users/components/ContactAvatar';
import { PrettyDate } from '../../layout/components/PrettyDate';

export interface IChatMessageProps {
  message: IConversationMessage;
}

export function ChatMessage({ message }: IChatMessageProps) {
  return (
    <Fragment>
      <ListItemAvatar>
        <ContactAvatar target={message.emitter} />
      </ListItemAvatar>
      <ListItemText
        primary={message.content}
        secondary={
          <span>
            <ContactName target={message.emitter} />
            {' - '}
            <PrettyDate date={message.createdAt} />
          </span>
        }
      />
    </Fragment>
  );
}
