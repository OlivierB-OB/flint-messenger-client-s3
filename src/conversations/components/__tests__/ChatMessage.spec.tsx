import React from 'react';
import { shallow } from 'enzyme';
import { ChatMessage } from '../ChatMessage';

describe('ChatMessage', () => {
  it('should display a message of a conversation', async () => {
    const message: any = {
      emitter: 'message emitter',
      content: 'message text',
      createdAt: 'message date',
    };
    const component = shallow(<ChatMessage message={message} />);
    expect(component).toMatchSnapshot();
  });
});
