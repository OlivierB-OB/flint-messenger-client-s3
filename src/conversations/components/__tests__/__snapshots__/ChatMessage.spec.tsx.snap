// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatMessage should display a message of a conversation 1`] = `
<Fragment>
  <WithStyles(ForwardRef(ListItemAvatar))>
    <Connect(UserAvatar)
      target="message emitter"
    />
  </WithStyles(ForwardRef(ListItemAvatar))>
  <WithStyles(ForwardRef(ListItemText))
    primary="message text"
    secondary={
      <span>
        <UNDEFINED
          target="message emitter"
        />
         - 
        <PrettyDate
          date="message date"
        />
      </span>
    }
  />
</Fragment>
`;
